servicetest/environment/test_environment.py

```diff
@@ -0,0 +1,276 @@
+
+# Copyright (C) 2016 Pelagicore AB
+#
+# Permission to use, copy, modify, and/or distribute this software for
+# any purpose with or without fee is hereby granted, provided that the
+# above copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+
+import pytest
+
+import os
+import time
+
+from testframework import Container
+
+import testhelper
+
+
+##### Useful globals #####
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+TESTOUTPUT_DIR = CURRENT_DIR + "/testoutput/"
+
+
+##### Provide what the testframework requires #####
+
+# This function is used by the 'agent' fixture to know where the log should be stored
+def logfile_path():
+    return TESTOUTPUT_DIR + "/environment-test.log"
+
+
+##### Local fixtures #####
+
+@pytest.fixture
+def clear_env_files(scope="function"):
+    """ Removes the files used by the helper to avoid having false
+        positives from previous test runs.
+
+        TODO: Should this be a method on the helper?
+    """
+    file_path = TESTOUTPUT_DIR + testhelper.ENV_VARS_FILE_NAME
+    if os.path.exists(file_path):
+        os.remove(file_path)
+
+
+@pytest.fixture
+def create_testoutput_dir(scope="module"):
+    """ Create a directory for the generated test files.
+
+        This directory is ignored by git but it's nice to have
+        somewhere locally to store test output to support
+        troubleshooting etc.
+    """
+    if not os.path.exists(TESTOUTPUT_DIR):
+        os.makedirs(TESTOUTPUT_DIR)
+
+
+##### Globals for setup and configuration of SC #####
+
+# The path to where the test app is located, will be passed in the 'data' dict when
+# starting the container. The test app is assumed to be located in the
+# same directory as this test.
+HOST_PATH = os.path.dirname(os.path.abspath(__file__))
+
+# These default values are used to pass various test specific values and
+# configurations to the Container helper. Tests that need to add, remove or
+# update entries can simply base their dict on this one for convenience.
+DATA = {
+    Container.PREFIX: "environment-test-",
+    Container.CONFIG: "[{\"enableWriteBuffer\": false}]",
+    Container.BIND_MOUNT_DIR: "app",
+    Container.HOST_PATH: HOST_PATH
+}
+
+
+#####  Various gateway configs used by the tests. #####
+#
+# The configs below might be dependent on each other so beware
+# when changing them.
+
+GW_CONFIG_ONE_VAR = [
```

Comment:
I think we should add another suite that tests the interaction between the GW configs and variables set explicitly when calling LaunchCommand, and I expect these configs to be reused then.